filament/src/materials/antiAliasing/taa.mat

```diff
@@ -0,0 +1,280 @@
+material {
+    name : taa,
+    parameters : [
+        {
+            type : sampler2d,
+            name : color
+        },
+        {
+            type : sampler2d,
+            name : depth,
+            precision: high
+        },
+        {
+            type : float,
+            name : alpha
+        },
+        {
+            type : sampler2d,
+            name : history
+        },
+        {
+            type : mat4,
+            name : reprojection,
+            precision: high
+        },
+        {
+            type : float[9],
+            name : filterWeights
+        }
+    ],
+    variables : [
+        vertex
+    ],
+    domain : postprocess,
+    depthWrite : false,
+    depthCulling : false
+}
+
+vertex {
+    void postProcessVertex(inout PostProcessVertexInputs postProcess) {
+        postProcess.vertex.xy = postProcess.normalizedUV;
+    }
+}
+
+fragment {
+
+// accurate box clipping
+#define BOX_CLIPPING_ACCURATE   0
+// clamping instead of clipping
+#define BOX_CLIPPING_CLAMP      1
+// no clipping (for debugging only)
+#define BOX_CLIPPING_NONE       2
+
+#if defined(TARGET_MOBILE)
+#define BOX_CLIPPING            BOX_CLIPPING_ACCURATE
+#define USE_YCoCg               0
+#define PREVENT_FLICKERING      0
+#define FILTER_HISTORY          0
+#else
+#define BOX_CLIPPING            BOX_CLIPPING_ACCURATE
+#define USE_YCoCg               1
+#define PREVENT_FLICKERING      0   // FIXME: thin lines disapear
+#define FILTER_HISTORY          1
+#endif
+
+// disable for debugging only
+#define HISTORY_REPORJECTION    1
+
+void dummy() {} // twart IDE
+
+float luma(const vec3 color) {
+#if USE_YCoCg
+    return color.x;
+#else
+#if defined(TARGET_MOBILE)
+    return max3(color);
+#else
+    return dot(color, vec3(0.2126, 0.7152, 0.0722));
```

Comment:
We have a built-in `luminance(vec3 x)` function you can call instead of rolling your own.